services/apis/indexd/indexdTasks.js

```diff
@@ -82,31 +86,77 @@ module.exports = {
    * @param {Object} fileNode - Assumed to have a did property
    * @returns {Promise<Gen3Response>}
    */
-  async getFile(file) {
+  async getFile(file, headers = user.mainAcct.accessTokenHeader) {
     // get data from indexd
     return I.sendGetRequest(
       `${indexdProps.endpoints.get}/${file.did}`,
-      user.mainAcct.accessTokenHeader,
+      headers,
     ).then((res) => {
       file.rev = getRevFromResponse(res);
       return res.body;
     });
   },
 
+  /**
+   * Updates indexd data for file
+   * @param {Object} fileNode - Assumed to have a did property
+   * @returns {Promise<Gen3Response>}
+   */
+  async updateFile(file, data, headers = user.mainAcct.accessTokenHeader) {
+    return I.sendGetRequest(
+      `${indexdProps.endpoints.get}/${file.did}`,
+      headers,
+    ).then((res) => {
+      // get last revision
+      file.rev = getRevFromResponse(res);
+
+      const headers = user.mainAcct.indexdAuthHeader;
+      headers['Content-Type'] = 'application/json; charset=UTF-8';
+      const strData = JSON.stringify(data);
+      return I.sendPutRequest(
+        `${indexdProps.endpoints.put}/${file.did}?rev=${file.rev}`,
+        strData,
+        headers,
+      ).then((res) => {
+        return res.body;
+      });
+    });
+  },
+
   /**
    * Deletes the file from indexd, given an indexd file object with did and rev
    * Response is added to the file object
    * @param {Object} file
    * @returns {Promise<Promise|*|PromiseLike<T>|Promise<T>>}
    */
-  async deleteFile(file) {
-    return I.sendDeleteRequest(
-      `${indexdProps.endpoints.delete}/${file.did}?rev=${file.rev}`,
-      user.mainAcct.indexdAuthHeader,
+  async deleteFile(file, headers = user.mainAcct.indexdAuthHeader) {
+    // if we already have the current revision we can use it, otherwise we need to get it
+    if (file.rev !== null && file.rev !== undefined) {
```

Comment:
why not something like this?
```
if (!file.rev) {
    file.rev = await I.sendGetRequest(...).then(res => getRevFromResponse(res));
}
return I.sendDeleteRequest(...);
```

Reply:
:+1:

Reply:
you shouldn't need to duplicate the `sendDeleteRequest` block

Reply:
ah, whoops, will fix